apps/projects/test/projects.test.js

```diff
@@ -89,72 +100,316 @@ contract('Projects App', accounts => {
+        assert.strictEqual(bounty2[5].toNumber(), 20)
+        await app.acceptFulfillment(repoId, 2, fulfillmentId2, { from: bountyAdder })
+        fulfillment2 = await registry.getFulfillment(bountyId2, fulfillmentId2)
+        assert(fulfillment2[0] === true)
+        bounty2 = await registry.getBounty(bountyId2)
+        assert.strictEqual(bounty2[5].toNumber(), 0)
+
+        const IssueData3 = await app.getIssue(repoId, 3)
+        const bountyId3 = IssueData3[1].toNumber()
+        const fulfillmentId3 = fulfilledBounty(
+          await registry.fulfillBounty(bountyId3, 'findthemillenniumfalcon')
+        )._fulfillmentId.toNumber()
+        let fulfillment3 = await registry.getFulfillment(bountyId3, fulfillmentId3)
+        assert(fulfillment3[0] === false)
+        let bounty3 = await registry.getBounty(bountyId3)
+        assert.strictEqual(bounty3[5].toNumber(), 30)
+        await app.acceptFulfillment(repoId, 3, fulfillmentId3, { from: bountyAdder })
+        fulfillment3 = await registry.getFulfillment(bountyId3, fulfillmentId3)
+        assert(fulfillment3[0] === true)
+        bounty3 = await registry.getBounty(bountyId3)
+        assert.strictEqual(bounty3[5].toNumber(), 0)
+      })
     })
   })
 
-  context('invalid operations', function() {
-    it('cannot retrieve a removed Repo', async function() {
-      repoID = (await app.addRepo('abc', String(123))).logs.filter(
-        x => x.event == 'RepoAdded'
-      )[0].args.id
-      await app.removeRepo(repoID, { from: repoRemover })
-      result = await app.getRepo(repoID)
+  context('invalid operations', () => {
+    it('cannot retrieve a removed Repo', async () => {
+      const repoId = addedRepo(
+        await app.addRepo('abc', String(123))
+      )
+      await app.removeRepo(repoId, { from: repoRemover })
+      const result = await app.getRepo(repoId)
       assert.equal(
         web3.toAscii(result[0]).replace(/\0/g, ''),
         '',
         'repo returned'
       )
     })
 
-    it('cannot add bounties to unregistered repos', function() {
+    it('cannot add bounties to unregistered repos', async () => {
       assertRevert(async () => {
         await app.addBounties('0xdeadbeef', [1, 2, 3], [10, 20, 30], {
           from: bountyAdder,
         })
       })
     })
+
+
+    it('cannot issue bulk bounties with invalid value', async () => {
+      const bountyAdder = accounts[2]
+      const repoId = addedRepo(
+        await app.addRepo('abc', String(123))
+      )
+      assertRevert(async () => {
+        await app.addBounties(
+          repoId,
+          [1, 2, 3],
+          [10, 20, 30],                   // 60 total Wei should be sent
+          [Date.now() + 86400, Date.now() + 86400, Date.now() + 86400],
+          [false, false, false],
+          [0, 0, 0],
+          'QmbUSy8HCn8J4TMDRRdxCbK2uCCtkQyZtY6XYv3y7kLgDCQmbUSy8HCn8J4TMDRRdxCbK2uCCtkQyZtY6XYv3y7kLgDCQmbUSy8HCn8J4TMDRRdxCbK2uCCtkQyZtY6XYv3y7kLgDC',
```

Comment:
I would use constants and concatenation to make it more clear what's happening here.